src/FunctionApps/python/IntakePipeline/__init__.py

```diff
@@ -1,46 +1,114 @@
 
 
-def run_pipeline():
+def run_pipeline(
+    bundle: dict,
+    datatype: str,
+    cred_manager: AzureFhirserverCredentialManager,
+):
     salt = get_required_config("HASH_SALT")
     geocoder = get_smartystreets_client(
         get_required_config("SMARTYSTREETS_AUTH_ID"),
         get_required_config("SMARTYSTREETS_AUTH_TOKEN"),
     )
-    fhirserver_cred_manager = get_fhirserver_cred_manager(
-        get_required_config("FHIR_URL")
+
+    transform_bundle(geocoder, bundle)
+    add_patient_identifier(salt, bundle)
+    store_bundle(
+        get_required_config("INTAKE_CONTAINER_URL"),
+        get_required_config("VALID_OUTPUT_CONTAINER_PATH"),
+        bundle,
+        datatype,
+    )
+    upload_bundle_to_fhir_server(cred_manager, bundle)
+
+
+def convert_to_fhir(
+    blob: func.InputStream,
+    fhir_url: str,
+    container_url: str,
+    invalid_output_path: str,
+    cred_manager: AzureFhirserverCredentialManager,
+) -> Tuple[str, dict]:
+    if blob.name[-3:].lower() not in ("hl7", "xml"):
+        raise Exception(f"invalid file extension for {blob.name}")
+
+    filetype = blob.name.split("/")[-2].lower()
+
+    if filetype == "elr":
+        message_type = "oru_r01"
+        message_format = "hl7v2"
+        bundle_type = "ELR"
+    elif filetype == "vxu":
+        message_type = "vxu_v04"
+        message_format = "hl7v2"
+        bundle_type = "VXU"
+    elif filetype == "eICR":
+        message_type = "ccda"
+        message_format = "ccda"
+        bundle_type = "ECR"
+    else:
+        raise Exception(f"Found an unidentified message_format: {filetype}")
+
+    # VA sends \\u000b & \\u001c in real data, ignore for now
+    messages = convert_batch_messages_to_list(
+        blob.read().decode("utf-8", errors="ignore")
     )
 
-    container_url = get_required_config("INTAKE_CONTAINER_URL")
-    container_prefix = get_required_config("INTAKE_CONTAINER_PREFIX")
-    output_path = get_required_config("OUTPUT_CONTAINER_PATH")
+    for i, message in enumerate(messages):
+        token = cred_manager.get_access_token()
+        response = convert_message_to_fhir(
+            message=message,
+            message_format=message_format,
+            message_type=message_type,
+            access_token=token.token,
+            fhir_url=fhir_url,
+        )
 
-    for datatype, bundle in read_fhir_bundles(container_url, container_prefix):
-        transform_bundle(geocoder, bundle)
-        add_patient_identifier(salt, bundle)
-        store_bundle(container_url, output_path, bundle, datatype)
-        upload_bundle_to_fhir_server(fhirserver_cred_manager, bundle)
+        if response.status_code == 200:
+            return bundle_type, response.json()
+        else:
+            ftype, fname = blob.name.split("/")[-2:]
+            fname, ext = fname.rsplit(".", 1)
+            filename = f"{ftype.lower()}-{fname}-{i}.{ext}"
+            store_message(container_url, invalid_output_path, filename, message)
```

Comment:
Not returning anything here and not raising an exception results in the code continuing to run on line 110, which results an unhandled exception where too many values are trying to be unpacked. Probably easiest to just return an empty tuple.

Reply:
Even correcting for this locally is still resulting in the unpacking error. I'm looking into why this is the case and will update once I know more.

Reply:
I would agree with the empty tuple return, and then we just define a check on the length of the tuple later on when processing bundles. Or, potentially, if we take the approach I outlined of building up a list of tuples, then we may not need to return anything at all here, since we'll already have a list of the fully valid and processed bundles.